src/app/retrospective/retrospective-list/retrospective-list.component.ts

```diff
@@ -30,26 +36,42 @@ export class RetrospectiveListComponent implements OnInit, OnDestroy {
     }
 
     ngOnInit() {
+        this.userService.getCurrentUser().takeUntil(this.destroy$).subscribe(
+            response => this.isAdmin = response.data.IsAdmin
+        );
+        this.showAll = this.retroDataService.getShowAllRetroState();
         this.initializeDataSource();
     }
-
     ngOnDestroy() {
         this.destroy$.next(true);
         this.destroy$.complete();
-        this.dataSource = null;
+        this.myRetroDataSource = null;
+        this.othersRetroDataSource = null;
     }
 
     initializeDataSource() {
-        this.dataSource = new RetrospectiveListDataSource(
-            this.retrospectiveService,
-            this.showCannotGetRetrospectivesError.bind(this)
-        );
+        this.retrospectiveService.getRetrospectives()
```

Comment:
please, read the documentation properly as you change the source of the table from `DataSource` to `array`.

Reply:
discussed